ngraph/src/ngraph/frontend/onnx_import/op/instance_norm.cpp

```diff
@@ -41,47 +41,89 @@ namespace ngraph
+                                     "Scale input must be one dimensional vector of number of "
+                                     "input data channels size.");
+
+                    CHECK_VALID_NODE(node,
+                                     (bias_pshape.rank().is_static() &&
+                                      bias_pshape.rank().get_length() == 1 &&
+                                      data_pshape.rank().is_static() &&
+                                      data_pshape[1].same_scheme(bias_pshape[0])),
                                      "Bias input must be one dimensional vector of number of "
                                      "input data channels size.");
 
                     // all dimensions except spatial/feature
-                    const AxisSet reduction_axes{
-                        common::get_monotonic_range<std::size_t>(data_shape.size(), 2)};
+                    const auto reduction_axes =
+                        common::get_monotonic_range_along_node_rank(data, 2);
 
                     const std::shared_ptr<ngraph::Node> eps_node =
                         std::make_shared<default_opset::Constant>(
-                            data->get_element_type(), data_shape, std::vector<float>{epsilon});
-
-                    scale = ngraph::builder::opset1::make_broadcast(scale, data_shape, 1);
-                    bias = ngraph::builder::opset1::make_broadcast(bias, data_shape, 1);
+                            data.get_element_type(), Shape{}, epsilon);
 
-                    Output<ngraph::Node> mean = builder::opset1::mean(data, reduction_axes);
-                    mean =
-                        ngraph::builder::opset1::make_broadcast(mean, data_shape, reduction_axes);
-
-                    Output<ngraph::Node> variance = builder::opset1::variance(data, reduction_axes);
-                    variance = ngraph::builder::opset1::make_broadcast(
-                        variance, data_shape, reduction_axes);
+                    Output<ngraph::Node> mean = builder::opset1::mean(data, reduction_axes, true);
+                    Output<ngraph::Node> variance =
+                        builder::opset1::variance(data, reduction_axes, true);
 
                     const auto sqrt = std::make_shared<default_opset::Sqrt>(
                         std::make_shared<default_opset::Add>(variance, eps_node));
 
+                    std::shared_ptr<ngraph::Node> data_shape_node;
+                    if (data_pshape.is_static())
+                    {
+                        data_shape_node = std::make_shared<default_opset::Constant>(
+                            element::i64,
+                            Shape{static_cast<size_t>(data_pshape.rank().get_length())},
+                            data_pshape.to_shape());
+                    }
+                    else
+                    {
+                        data_shape_node = std::make_shared<default_opset::ShapeOf>(data);
```

Comment:
I think we can't reach this `else` because of `data_pshape.rank().is_static()` is required in the `CHECK_VALID_NODE`.